joblib/pool.py

```diff
@@ -428,10 +427,23 @@ def _setup_queues(self):
         self._quick_get = self._outqueue._recv
 
 
-def delete_folder(folder_path):
+def delete_folder(folder_path, n_retries=10, delay=0.1):
     """Utility function to cleanup a temporary folder if still existing"""
-    if os.path.exists(folder_path):
-        shutil.rmtree(folder_path)
+    i = 0
+    while i < n_retries:
+        try:
+            if os.path.exists(folder_path):
+                shutil.rmtree(folder_path)
+            break
+        except WindowsError:
+            # A worker process might still hold an open file descriptors
+            # on one of the memory mapped arrays in the temporary folder
+            # let's wait a bit an retry
+            if i >= n_retries:
+                raise
+            else:
+                i += 1
+                sleep(delay)
```

Comment:
I think you would never raise because i < n_retries. Also a for loop may be a tad simpler.

```python
for i in range(n_retries)
    try:
        if os.path.exists(folder_path):
            shutil.rmtree(folder_path)
        break
    except WindowsError:
        # A worker process might still hold an open file descriptors
        # on one of the memory mapped arrays in the temporary folder
        # let's wait a bit an retry
        if i >= n_retries - 1:
            raise
        else:
            sleep(delay)
```